x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -183,7 +183,10 @@
                             .privileges("read", "view_index_metadata").build(),
                         RoleDescriptor.IndicesPrivileges.builder().indices("observability-annotations")
                             .privileges("read", "view_index_metadata").build()
-                    }, null, MetadataUtils.DEFAULT_RESERVED_METADATA))
+                    }, new RoleDescriptor.ApplicationResourcePrivileges[] {
+                            RoleDescriptor.ApplicationResourcePrivileges.builder()
+                                .application("kibana-*").resources("*").privileges("reserved_ml_apm_user").build()
```

Comment:
Clarifying question since I'm not familiar with "reserved" privileges: I couldn't find `reserved_ml_apm_user` in https://github.com/elastic/kibana/pull/72266 but I did find `ml_apm_user`. Is `reserved_`  an implied prefix somehow?

Reply:
Good question. Yes, the `reserved_` prefix is generated by Kibana's security plugin when it "serializes" our privilege model into Elasticsearch's Application Privilege model:

https://github.com/elastic/kibana/blob/bf04235dae35452061cc7ea3d86d96c19a58206c/x-pack/plugins/security/server/authorization/privilege_serializer.ts#L61